tests/spec/combine_series/test_event_series_and_event_series.py

```diff
@@ -1,36 +1,23 @@
-from databuilder.query_language import Dataset, IdSeries, IntSeries, build_event_table
+from ..tables import e
 
-from ..helpers import transpose
+table_data = {
+    e: """
+          |  i1 |  i2
+        --+-----+-----
+        1 | 101 | 111
+        1 | 102 | 112
+        2 | 201 | 211
+        2 | 202 | 212
+    """,
+}
 
-e = build_event_table(
-    "e",
-    {
-        "patient_id": IdSeries,
-        "i1": IntSeries,
-        "i2": IntSeries,
-    },
-)
 
-
-def test_combine_event_series_and_event_series(in_memory_engine):
-    in_memory_engine.setup(
+def test_event_series_and_event_series(spec_test):
+    spec_test(
+        table_data,
+        (e.i1 + e.i2).sum_for_patient(),
```

Comment:
It seems a bit of a shame that all of these `combine_series` tests are dependent on both integer arithmetic and `sum_for_patient()`. But I imagine you've come to the conclusion that this is the simplest series you can define that "works" for all the cases you want to cover. And I think you're right to favour making them uniform rather then simplifying the ones that don't involve event series.

It would be nice (especially once we get to generating doc from this) to have some overarching text explaining that these specs show the combination behaviour and that the arithmetic details aren't the important thing. We can defer that to the generate doc story; I think we'll either need to have a way of providing module-level text or merge multiple specs into one file.

Reply:
Agree with all of this.